pkg/cluster/delete.go

```diff
@@ -425,13 +427,12 @@ func (m *manager) deleteFederatedCredentials(ctx context.Context) error {
 				&armmsi.FederatedIdentityCredentialsClientDeleteOptions{},
 			)
 			if err != nil {
-				cloudErr := err.(*api.CloudError)
-
-				if cloudErr.StatusCode == http.StatusNotFound {
-					m.log.Errorf("federated identity credentials not found for %s: %v", identity.ResourceID, cloudErr.Error())
+				var respErr *azcore.ResponseError
+				if errors.As(err, &respErr); respErr.StatusCode == http.StatusNotFound {
```

Comment:
You can use this.
https://github.com/Azure/ARO-RP/blob/2143cb81353867aa5ff21d15dfcc4f51cc8c16c2/pkg/util/azureclient/azuresdk/errors/errors.go#L14-L17

Reply:
https://github.com/Azure/ARO-RP/pull/3915